CARDS-2737: Improve the UI for the patient portal page layout

veronikaslc opened the pull request:
Specs: [CARDS-2737](https://phenotips.atlassian.net/browse/CARDS-2737)

**To test the YE workflows:**
* start with `-P prems`
* before authenticating, navigate to /Survey.html to check the default patient portal screen
* as admin create a Patient subject with a Patient information form and a Visit subject with a visit information form
  * obtain the visit token from `/Subjects/<PATIENT UUID>/<VISIT UUID>.token.html`
* in a different browser or in incognito mode, navigate to `/Survey.html?auth_token=<THE TOKEN>`
  * complete the survey to test all YE screens
  * repeat the test for different YE surveys to check the welcome screen layout with different intros and numbers of questionnaires
  * click on Unsubscribe in the footer to test the unsubscribe/resubscribe screens and functionality

**To test other workflows:**
* as admin navigate to Administration > Patient Access
  * tick the checkbox "_Patients can answer surveys without a personalized link_" and save
  * in a different browser or in incognito mode, navigate to /Survey.html? and check the layout of the patient identification page
* as admin, on  the Administration > Patient Access page, enter a negative number in the box labeled "_Relatively to the associated event, patients can start submitting incomplete surveys within:_" to enable the submission of incomplete surveys and save
  * go through the YE patient workflow **without answering any of the questions**
  * check the layout of the "Incomplete surveys" screen just before the Review step


[CARDS-2737]: https://phenotips.atlassian.net/browse/CARDS-2737?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

veronikaslc committed:
CARDS-2737: Improve the UI for the patient portal page layout

marta- commented on modules/patient-portal/src/main/frontend/src/patient-portal/QuestionnaireSet.jsx:
```diff
@@ -865,7 +866,15 @@ function QuestionnaireSetScreen (props) {
+      {Array.from(children || []).filter(c => c).map((c, i) =>
+          <Grid
+            key={i+"MainItem"}
+            alignSelf={["welcome-action", "expiry-message", "review-title", "summary-title"].includes(c.key) || c.key.startsWith("review-submit") ? "center" : ""}
```
WSORD at the review screen, seems related to this line, to be confirmed.

<img width="1054" height="106" alt="image" src="https://github.com/user-attachments/assets/92cda97b-a22b-4af3-b8c9-79d77e9c65c6" />

marta- commented on modules/patient-portal/src/main/frontend/src/patient-portal/QuestionnaireSet.jsx:
```diff
@@ -663,19 +664,19 @@ function QuestionnaireSet(props) {
+    <List key="welcome-surveys" sx={{p : 0}}>
     { (questionnaireIds || []).map((q, i) => (
-      <ListItem key={q+"Welcome"}>
+      <ListItem key={q+"Welcome"} sx={{pt : 0, pb: 0}}>
```
`py` should work iirc.

```suggestion
      <ListItem key={q+"Welcome"} sx={{py : 0}}>
```

marta- replied:
While this styling is according to the initial mockup, it looks neater with disablePadding:

<img width="747" height="1328" alt="image" src="https://github.com/user-attachments/assets/aa073df7-73de-479e-9d4d-271f3d50626b" />

versus

<img width="722" height="1320" alt="image" src="https://github.com/user-attachments/assets/eb1c18b8-e5f9-48f2-aca7-ec6b2b58fe32" />

marta- commented on modules/patient-portal/src/main/frontend/src/patient-portal/QuestionnaireSet.jsx:
```diff
@@ -591,7 +591,8 @@ function QuestionnaireSet(props) {
     let timeOfDay = hourOfDay < 12 ? "morning" : hourOfDay < 18 ? "afternoon" : "evening";
-    return `Good ${timeOfDay}` + (name ? `, ${name}` : '');
+    let greeting = `Good ${timeOfDay}` + (name ? `, ${name}` : '');
+    return <Typography variant="h6" key="welcome-greeting" sx={{fontWeight: "600"}}>{ greeting }</Typography>;
```
Is the fontWeight necessary? I don't see a difference without it (Firefox/Win11).

veronikaslc replied:
I see in Chrome but if it's not critical than let's remove

marta- commented on modules/patient-portal/src/main/frontend/src/patient-portal/QuestionnaireSet.jsx:
```diff
@@ -663,19 +664,19 @@ function QuestionnaireSet(props) {
+      : displayText("surveyIntro", Typography, {key: "welcome-message"})
     ),
-    <List key="welcome-surveys">
+    <List key="welcome-surveys" sx={{p : 0}}>
```
Could the `dense` prop be used instead of custom styling?

veronikaslc replied:
It changes font to smaller so I decided to use custom styling, but lets see it it looks better with dense.

marta- replied:
I see. `dense` is not exactly what we want, but `disablePadding` on both `<List>` and `<ListItem>` should do the trick.

veronikaslc replied:
If we disablePadding on the `<ListItem>` then it will loose the left padding and alignment will not look like in the mockup. I'll use `sx={{py : 0}}` for it instead.

marta- replied:
While this styling is according to the initial mockup, it looks neater with disablePadding:

<img width="747" height="1328" alt="image" src="https://github.com/user-attachments/assets/aa073df7-73de-479e-9d4d-271f3d50626b" />

versus

<img width="722" height="1320" alt="image" src="https://github.com/user-attachments/assets/eb1c18b8-e5f9-48f2-aca7-ec6b2b58fe32" />

marta- reviewed (dismissed):
The gray background from the specs is missing, but we can leave it out in this iteration.

veronikaslc committed:
CARDS-2737: Improve the UI for the patient portal page layout
Addressed codereview comments

marta- commented:
Let's also left-align the page displayed when there's no auth_token:

<img width="915" height="947" alt="image" src="https://github.com/user-attachments/assets/50f30a00-f504-4f1f-8aac-ae74fc3bf173" />

veronikaslc commented:
What about [unsubscribe](https://github.com/data-team-uhn/cards/blob/dev/modules/patient-portal/src/main/frontend/src/patient-portal/unsubscribe.jsx) page?

marta- commented on modules/patient-portal/src/main/frontend/src/patient-portal/QuestionnaireSet.jsx:
```diff
@@ -591,7 +591,8 @@ function QuestionnaireSet(props) {
     let timeOfDay = hourOfDay < 12 ? "morning" : hourOfDay < 18 ? "afternoon" : "evening";
-    return `Good ${timeOfDay}` + (name ? `, ${name}` : '');
+    let greeting = `Good ${timeOfDay}` + (name ? `, ${name}` : '');
+    return <Typography variant="h6" key="welcome-greeting">{ greeting }</Typography>;
```
We should make this configurable.

1. In SurveyInstructionsConfiguration, under **Start Screen**, right after "Enable start screen" add a text field for "Greeting". If present, display that greeting, otherwise default to the one above.
<img width="847" height="1262" alt="image" src="https://github.com/user-attachments/assets/8f2a2f1e-0620-401e-b916-aaf1ff8f441f" />


2. **In a separate commit**, set the greeting field to be "Your Experience at UHN" for prems in https://github.com/data-team-uhn/cards/blob/3396a617dd60038df5ff1d235739e006bbca5355/prems-resources/clinical-data/src/main/resources/SLING-INF/content/Survey/SurveyInstructions.xml .

marta- commented:
> What about [unsubscribe](https://github.com/data-team-uhn/cards/blob/dev/modules/patient-portal/src/main/frontend/src/patient-portal/unsubscribe.jsx) page?

Good catch.

Similar layout to the above: narrow content container centered on the page, logo and text left-aligned, button(s) right aligned.

veronikaslc committed:
CARDS-2737: Improve the UI for the patient portal page layout
Adjusted more patient portal pages to be left aligned
Added greeting to the Survey configuration

veronikaslc committed:
CARDS-2737: Improve the UI for the patient portal page layout
Set the greeting field for prems

veronikaslc committed:
CARDS-2737: Improve the UI for the patient portal page layout
Optimise padding for lists

marta- commented:
Minor, but when the review screen is loading the circular progress is now left-aligned instead of centered and looks really out of place.

marta- commented:
Also minor: on the Unsubscribe page, the width of the content differs before and after clicking UNSUBSCRIBE, so the logo and the button appear to change position which looks awkward. It wasn't noticeable with center alignment, but it is now with left alignment. Can be improved by having a specified max-width for the content.

<img width="665" height="445" alt="image" src="https://github.com/user-attachments/assets/b8cd7546-3ff3-4fff-bffc-f23ff29d4cf1" />

<img width="667" height="502" alt="image" src="https://github.com/user-attachments/assets/ff522f94-48af-4e5f-b52f-93d48a1e375d" />

veronikaslc committed:
CARDS-2737: Improve the UI for the patient portal page layout
Better central positioning of loading element and unsubscribe container

veronikaslc commented:
Circular progress fix is comming

veronikaslc committed:
CARDS-2737: Improve the UI for the patient portal page layout
Disable padding for ListItems as well

veronikaslc committed:
CARDS-2737: Improve the UI for the patient portal page layout
Better central positioning of all loading elements

marta- committed:
CARDS-2737: Improve the UI for the patient portal page layout
Better central positioning of all loading elements - refactored

marta- committed:
CARDS-2737: Improve the UI for the patient portal page layout
Removed the line break from the default draft info survey instructions because it works better with left alignment.

marta- commented:
Made two small changes that diverge slightly from the original specs:
* Left-aligned expiry message, so that only the action is centered on the welcome screen
* Removed the newline from the draft info portion for a better look in the left-aligned layout.

marta- committed:
CARDS-2737: Improve the UI for the patient portal page layout
Better font sizes and styles on the welcome screen

marta- committed:
CARDS-2737: Improve the UI for the patient portal page layout
- Refactoring to accomodate the styling of the incomplete surveys screen
- Reorganization of the incomplete surveys screen:
  - Remove survey list padding
  - Use alert for the message
  - Right-align actions
  - Place the primary action first from the right

marta- committed:
CARDS-2737: Improve the UI for the patient portal page layout
Unsubscribe page:
- Ensure the content width is the same regardless of the message length
- Use info alert with no icon for the initial unsubscribe message
- Use success alert when unsubscribe/resubscribe action is successful

marta- committed:
CARDS-2737: Improve the UI for the patient portal page layout
YE surveys: removed the newlines from survey intros around the visit location.

marta- committed:
CARDS-2737: Improve the UI for the patient portal page layout
Unsubscribe page cleanup:
- Removed unused Typography import
- Reorganized MUI imports

marta- committed:
CARDS-2737: Improve the UI for the patient portal page layout
Patient identification page: added ":" after "Enter the following information for identification" and make it stand out more.

marta- committed:
[misc] Better spacing of the MRN and HC boxes on the patient identification page on narrow screens

Hajany commented:
I believe Sergiu brought this up before, can we make the free text box wider to fit the question box? 
<img width="846" height="421" alt="image" src="https://github.com/user-attachments/assets/f357a34e-530a-4fd8-8fa6-a2d82e85f26e" />

marta- commented:
> I believe Sergiu brought this up before, can we make the free text box wider to fit the question box? 

We can change its size, however it is out of scope for this PR.

Are we sure we want to make it as wide as the question container? It will look really large on wide screens.

Hajany commented:
> > I believe Sergiu brought this up before, can we make the free text box wider to fit the question box?
> 
> We can change its size, however it is out of scope for this PR.
> 
> Are we sure we want to make it as wide as the question container? It will look really large on wide screens.

maybe half the size of the question container?
